x/feemarket/types/resolver.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// DenomResolver is an interface to convert a given token to the feemarket's base token.
+type DenomResolver interface {
+	// ConvertToBaseToken converts feeCoin into the equivalent amount of the token denominated in baseDenom.
```

Comment:
Does this imply that the converted fee ends up being deducted from the users account or that we just do a sanity check that the base fee is met for the tx?

Reply:
Yeah this is just a sanity check.

`if( converted) >= required); then deduct(suppliedFee)